src/Plugin.php

```diff
@@ -86,6 +93,45 @@ public static function upload_mime_types(array $mimes) {
   }
 
   /**
+   * Removes SVG image size attributes from the content.
+   *
+   * @implements the_content
+   */
+  public static function the_content($content) {
+    $content = preg_replace_callback('@<img [^>]*src=[^>]+?.svg[^>]*>@', function ($matches) {
+      return static::removeSizeAttributes($matches[0]);
+    }, $content);
+    return $content;
+  }
+
+  /**
+   * Removes size attributes from SVG images.
+   *
+   * @implements wp_get_attachment_metadata
+   * @implements image_send_to_editor
+   * @implements post_thumbnail_html
+   */
+  public static function removeSvgSizeAttributes($data, $post_id, $attachment_id = NULL) {
+    if ($data && get_post_mime_type($attachment_id ?? $post_id) === 'image/svg+xml') {
```

Comment:
Yes, this is why I'm registering the filter with 2 possible arguments only to not getting the caption in there:
`add_filter('image_send_to_editor', __CLASS__ . '::removeSvgSizeAttributes', 10, 2);`